pkg/kv/kvserver/client_replica_test.go

```diff
@@ -5259,7 +5263,8 @@ func TestOptimisticEvalNoContention(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
```

Comment:
If you wanted, you could use `testutils.SucceedsSoonDuration()` for the timeout. It is 45 seconds by default though, but has the advantage of being scaled under race/deadlock.  Although, I imagine even under race/deadlock we expect this to take much less than 15 seconds.